### PR TITLE
number challenge update

### PR DESCRIPTION
**Pull request by susanharper-okta**
## Description:
- **What's changed?** Added number challenge Okta Verify Push support for MFA OOB and OOB 
- **Is this PR related to a Monolith release?** Yes 2023.09.0

Netlify links:

MFA OOB
https://64efcc638b4b1516421f7a4b--reverent-murdock-829d24.netlify.app/docs/guides/implement-grant-type/mfaoob/main/

OOB
https://64efcc638b4b1516421f7a4b--reverent-murdock-829d24.netlify.app/docs/guides/implement-grant-type/oob/main/

### Resolves:

* [OKTA-640600](https://oktainc.atlassian.net/browse/OKTA-640600)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/guides/implement-grant-type/main/mfaoob/use-flow.md [hunk 1]
```diff
@@ -80,14 +80,29 @@ In an HTTP 200 response, Okta returns the following parameters:
...
+- `challenge_type`: The challenge type used for authentication. This must be one of the `challenge_types_supported` from the request body. The MFA OOB flow supports only `http://auth0.com/oauth/grant-type/mfa-oob`.
+- `oob_code`: An identifier of an out-of-band factor transaction. To respond to or check on the status of an out-of-band factor, this code is used to identify the factor transaction.
+- `expires_in`: The time, in seconds, until the `oob_code` expires
+- `interval`: The frequency, in seconds, at which the client should poll to check if the out-of-band factor is complete. This is only relevant to polling factors such as Okta Verify Push.
+- `channel`: The type of out-of-band channel used. Okta currently only supports Okta Verify Push.<!-- need to update this when phase 2 is complete -->
+- `binding_method`: The method used to bind the out-of-band channel with the primary channel. Supported values: `none`, `transfer`. Use `transfer` when you configure the number challenge for Okta Verify Push.
```
Comment (dannavarro-okta) [inline packages/@okta/vuepress-site/docs/guides/implement-grant-type/main/mfaoob/use-flow.md:104]: Can please you remove the last sentence? “Use `transfer` when you configure the number challenge for Okta Verify Push.“? The server returns the `binding_method` and to me it sounds like the customer would have to be doing something when that is not the case.

File: packages/@okta/vuepress-site/docs/guides/implement-grant-type/main/oob/use-flow.md [hunk 2]
```diff
@@ -40,13 +40,27 @@ In an HTTP 200 response, Okta returns the following parameters:
...
-- `binding_method` is the method used to bind the out-of-band channel with the primary channel. Since Okta Verify Push doesn't require any binding, the only supported value for EA is `none`.<!-- need to update this when phase 2 is complete -->
+- `oob_code`: An identifier of a single out-of-band factor transaction. To respond to or check on the status of an out-of-band factor, this code is used to identify the factor transaction.
+- `expires_in`: The time, in seconds, until the `oob_code` expires
+- `interval`: The frequency, in seconds, at which the client needs to poll Okta to check if the out-of-band factor is completed. This is only relevant to polling factors such as Okta Verify Push.
+- `channel`: The type of out-of-band channel used. Okta currently only supports Okta Verify Push.<!-- need to update this when phase 2 is complete -->
+- `binding_method`: The method used to bind the out-of-band channel with the primary channel. Supported values: `none`, `transfer`. Use `transfer` when you configure the number challenge for Okta Verify Push.
```
Comment (dannavarro-okta) [inline packages/@okta/vuepress-site/docs/guides/implement-grant-type/main/oob/use-flow.md:62]: Can please you remove the last sentence? “Use `transfer` when you configure the number challenge for Okta Verify Push.“? The server returns the `binding_method` and to me it sounds like the customer would have to be doing something when that is not the case.

**Comment by dannavarro-okta** [comment]
one minor comment but otherwise looks good. thanks

**Review by dannavarro-okta — APPROVED** [review]

File: packages/@okta/vuepress-site/docs/guides/implement-grant-type/main/mfaoob/flow-diagram.md [hunk 3]
```diff
@@ -39,10 +41,21 @@ At a high level, this flow has the following steps:
 
 1. Okta sends an HTTP 403 error that MFA is required and includes the `mfa_token` in the response.
 1. Your app sends a `/challenge` request with the `mfa_token` value and the `challenge_types_supported` (`http://auth0.com/oauth/grant-type/mfa-oob`) to the Okta authorization server.
-1. Okta responds with a `challenge_type`, `oob_code`, and `interval` in seconds to poll (default is `5`).
+1. Okta responds with a `challenge_type`, `oob_code`, `interval` in seconds to poll (default is `5`), and any other parameters required by the configured authenticator.
+
+   >**Note:** For Okta Verify prompts with [number challenge](https://help.okta.com/okta_help.htm?type=oie&id=ext-config-okta-verify-options), `binding_method=transfer` and `binding_code` are also returned. See the [challenge response example](#challenge-response).
+
 1. Okta sends a push notification to the user.
+1. Per configured authenticator options, more interaction may occur:
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/guides/implement-grant-type/main/mfaoob/flow-diagram.md:49]:
```suggestion
1. Per configured authenticator options, more interaction may occur.
```
No bullet list?

File: packages/@okta/vuepress-site/docs/guides/implement-grant-type/main/mfaoob/flow-diagram.md [hunk 4]
```diff
@@ -39,10 +41,21 @@ At a high level, this flow has the following steps:
...
+
+   >**Note:** For Okta Verify prompts with number challenge, the client displays the `binding_code` to the user.
+
 1. Your app polls the Okta `/token` endpoint at the set `interval`. The `oob_code`, `mfa_token`, and [MFA OOB grant type](https://developer.okta.com/docs/api/openapi/okta-oauth/oauth/tag/OrgAS/#tag/OrgAS/operation/token) (`grant_type=http://auth0.com/oauth/grant-type/mfa-oob`) are included in the requests.
 1. Okta responds with an HTTP 400 `authorization_pending` error.
-1. The user responds to the push notification.
+1. The user opens the Okta Verify app and taps **Yes it's me**.
+1. Per configured authenticator options, more interaction may occur:
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/guides/implement-grant-type/main/mfaoob/flow-diagram.md:56]:
```suggestion
1. Per configured authenticator options, more interaction may occur.
```
If you want to keep the colon, maybe don't put the next sentence in a Note?

File: packages/@okta/vuepress-site/docs/guides/implement-grant-type/main/mfaoob/use-flow.md [hunk 5]
```diff
@@ -80,14 +80,29 @@ In an HTTP 200 response, Okta returns the following parameters:
...
-- `channel` is the type of out-of-band channel used. Okta currently only supports Okta Verify Push.<!-- need to update this when phase 2 is complete -->
-- `binding_method` is the method used to bind the out-of-band-channel with the primary channel. Since Okta Verify Push doesn't require any binding, the only supported value for EA is `none`.<!-- need to update this when phase 2 is complete -->
+- `challenge_type`: The challenge type used for authentication. This must be one of the `challenge_types_supported` from the request body. The MFA OOB flow supports only `http://auth0.com/oauth/grant-type/mfa-oob`.
+- `oob_code`: An identifier of an out-of-band factor transaction. To respond to or check on the status of an out-of-band factor, use this code to identify the factor transaction.
+- `expires_in`: The time, in seconds, until the `oob_code` expires
+- `interval`: The frequency, in seconds, at which the client should poll to check if the out-of-band factor is complete. This is only relevant to polling factors such as Okta Verify Push.
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/guides/implement-grant-type/main/mfaoob/use-flow.md:102]:
```suggestion
- `interval`: The frequency, in seconds, that the client needs to poll to check if the out-of-band factor is complete. This is only relevant to polling factors such as Okta Verify Push.
```

File: packages/@okta/vuepress-site/docs/guides/implement-grant-type/main/oob/flow-diagram.md [hunk 6]
```diff
@@ -17,28 +17,41 @@ autonumber "<b>#."
...
 
     Register your app so that Okta can accept the authorization request. See [Set up your app](#set-up-your-app) to register and configure your app with Okta. After registration, your app can make an authorization request to Okta. See [Request for tokens](#request-for-tokens).
 
-1. Okta returns an acknowledgment, the `oob_code`, and the `interval` in seconds to poll (default is `5`).
+1. Okta responds with the `oob_code`, `interval` in seconds to poll (default is `5`), and any other parameters required by the configured authenticator.
+
+   >**Note:** For Okta Verify prompts with [number challenge](https://help.okta.com/okta_help.htm?type=oie&id=ext-config-okta-verify-options), `binding_method=transfer` and `binding_code` are also returned. See the [OOB response example](#oob-response-example).
+
 1. Okta sends a push notification to the user.
+1. Per configured authenticator options, more interaction may occur:
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/guides/implement-grant-type/main/oob/flow-diagram.md:45]:
```suggestion
1. Per configured authenticator options, more interaction may occur.
```

File: packages/@okta/vuepress-site/docs/guides/implement-grant-type/main/oob/flow-diagram.md [hunk 7]
```diff
@@ -17,28 +17,41 @@ autonumber "<b>#."
...
+
+   >**Note:** For Okta Verify prompts with number challenge, the client displays the `binding_code` to the user.
+
 1. Your app polls the Okta `/token` endpoint at the set `interval`. The `oob_code` and the [OOB grant type](https://developer.okta.com/docs/api/openapi/okta-oauth/oauth/tag/OrgAS/#tag/OrgAS/operation/token) (`grant_type=urn:okta:params:oauth:grant-type:oob`) are included in the requests.
 1. Okta responds with an HTTP 400 `authorization_pending` error.
-1. The user responds to the push notification.
+1. The user opens the Okta Verify app and taps **Yes it's me**.
+1. Per configured authenticator options, more interaction may occur:
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/guides/implement-grant-type/main/oob/flow-diagram.md:52]:
```suggestion
1. Per configured authenticator options, more interaction may occur.
```

**Review by vanngo-okta — APPROVED** [review]
LGTM! (Just a few style suggestions for you to review)

File: packages/@okta/vuepress-site/docs/guides/implement-grant-type/main/mfaoob/flow-diagram.md (hunk 3 above)
Comment (susanharper-okta) [reply packages/@okta/vuepress-site/docs/guides/implement-grant-type/main/mfaoob/flow-diagram.md:49]: not yet - there's only one right now. I'll do the period until the next changes come and it requires a bulleted list. Good catch.

File: packages/@okta/vuepress-site/docs/guides/implement-grant-type/main/mfaoob/flow-diagram.md (hunk 4 above)
Comment (susanharper-okta) [reply packages/@okta/vuepress-site/docs/guides/implement-grant-type/main/mfaoob/flow-diagram.md:56]: see above comment

**Pull request merged**